### ISSUE TITLE
question about training the baseline

### ISSUE THREAD

**Issue by cindybrain** [NervanaSystems/distiller#445]
I am so confused why I use the same command to train the baseline, but I get a different result.
Is there anything else I need to concerned about?

command:
time python3 compress_classifier.py --arch vgg16_cifar  ../../../data.cifar10 -p=50 --lr=0.05 --epochs=180 --compress=../baseline_networks/cifar/vgg16_cifar_baseline_training.yaml -j=1 --deterministic

Result:
Parameters:
+----+---------------------------+------------------+---------------+----------------+------------+------------+----------+----------+----------+------------+---------+----------+------------+
|    | Name                      | Shape            |   NNZ (dense) |   NNZ (sparse) |   Cols (%) |   Rows (%) |   Ch (%) |   2D (%) |   3D (%) |   Fine (%) |     Std |     Mean |   Abs-Mean |
|----+---------------------------+------------------+---------------+----------------+------------+------------+----------+----------+----------+------------+---------+----------+------------|
|  0 | features.module.0.weight  | (64, 3, 3, 3)    |          1728 |           1728 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.07259 | -0.00074 |    0.06301 |
|  1 | features.module.2.weight  | (64, 64, 3, 3)   |         36864 |          36864 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01544 |  0.00009 |    0.01336 |
|  2 | features.module.5.weight  | (128, 64, 3, 3)  |         73728 |          73728 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01550 | -0.00001 |    0.01341 |
|  3 | features.module.7.weight  | (128, 128, 3, 3) |        147456 |         147456 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01097 |  0.00000 |    0.00950 |
|  4 | features.module.10.weight | (256, 128, 3, 3) |        294912 |         294912 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01096 |  0.00002 |    0.00949 |
|  5 | features.module.12.weight | (256, 256, 3, 3) |        589824 |         589824 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00775 | -0.00000 |    0.00671 |
|  6 | features.module.14.weight | (256, 256, 3, 3) |        589824 |         589824 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00775 | -0.00001 |    0.00671 |
|  7 | features.module.17.weight | (512, 256, 3, 3) |       1179648 |        1179648 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00775 |  0.00001 |    0.00671 |
|  8 | features.module.19.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00548 |  0.00000 |    0.00475 |
|  9 | features.module.21.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00548 | -0.00001 |    0.00475 |
| 10 | features.module.24.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00548 | -0.00000 |    0.00475 |
| 11 | features.module.26.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00548 |  0.00000 |    0.00474 |
| 12 | features.module.28.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00548 |  0.00000 |    0.00475 |
| 13 | classifier.weight         | (10, 512)        |          5120 |           5120 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01589 |  0.00021 |    0.01370 |
| 14 | Total sparsity:           | -                |      14715584 |       14715584 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00000 |  0.00000 |    0.00000 |
+----+---------------------------+------------------+---------------+----------------+------------+------------+----------+----------+----------+------------+---------+----------+------------+
2019-12-13 14:36:46,232 - Total sparsity: 0.00

2019-12-13 14:36:46,232 - --- validate (epoch=179)-----------
2019-12-13 14:36:46,232 - 5000 samples (256 per mini-batch)
2019-12-13 14:37:00,206 - ==> Top1: 9.460    Top5: 48.320    Loss: 2.303

2019-12-13 14:37:00,245 - ==> Best [Top1: 10.540   Top5: 49.460   Sparsity:0.00   NNZ-Params: 14715584 on epoch: 42]
2019-12-13 14:37:00,246 - Saving checkpoint to: logs/2019.12.12-230254/checkpoint.pth.tar
2019-12-13 14:37:00,461 - --- test ---------------------
2019-12-13 14:37:00,461 - 10000 samples (256 per mini-batch)
2019-12-13 14:37:28,288 - ==> Top1: 10.000    Top5: 50.000    Loss: 2.303

**Comment by nzmora** [comment]
Hi @cindybrain,

I just ran the same command-line you provided above and you can see that the network is learning quite fast (Top1: 73.2 after 7 epochs).

Your network is not learning anything (Top1:10 is 10% correct, which is the expected value of a uniform-random distribution over 10 classes).  Take a look at the log files to see what happens in the early epochs: 

- is the learning-rate (LR) set correctly (0.05 at first)?
- is the LR changing as defined in the YAML file?
- is the batch size set to 256?
- is the training loss getting smaller? 

Please check that you're CIFAR10 dataset is correct (perhaps download it again).
Please check that you haven't changed the YAML schedule file `../baseline_networks/cifar/vgg16_cifar_baseline_training.yaml`.

If you need, I can attach a log file of my session.

Good luck,
Neta

```
Parameters:
+----+---------------------------+------------------+---------------+----------------+------------+------------+----------+----------+----------+------------+---------+----------+------------+
|    | Name                      | Shape            |   NNZ (dense) |   NNZ (sparse) |   Cols (%) |   Rows (%) |   Ch (%) |   2D (%) |   3D (%) |   Fine (%) |     Std |     Mean |   Abs-Mean |
|----+---------------------------+------------------+---------------+----------------+------------+------------+----------+----------+----------+------------+---------+----------+------------|
|  0 | features.module.0.weight  | (64, 3, 3, 3)    |          1728 |           1728 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.15165 | -0.00028 |    0.11688 |
|  1 | features.module.2.weight  | (64, 64, 3, 3)   |         36864 |          36864 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.06501 | -0.00635 |    0.05162 |
|  2 | features.module.5.weight  | (128, 64, 3, 3)  |         73728 |          73728 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.04709 | -0.00391 |    0.03724 |
|  3 | features.module.7.weight  | (128, 128, 3, 3) |        147456 |         147456 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.04343 | -0.00432 |    0.03469 |
|  4 | features.module.10.weight | (256, 128, 3, 3) |        294912 |         294912 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.03131 | -0.00203 |    0.02491 |
|  5 | features.module.12.weight | (256, 256, 3, 3) |        589824 |         589824 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.02934 | -0.00143 |    0.02341 |
|  6 | features.module.14.weight | (256, 256, 3, 3) |        589824 |         589824 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.02925 | -0.00228 |    0.02338 |
|  7 | features.module.17.weight | (512, 256, 3, 3) |       1179648 |        1179648 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.02067 | -0.00091 |    0.01649 |
|  8 | features.module.19.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01999 | -0.00053 |    0.01596 |
|  9 | features.module.21.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01989 | -0.00065 |    0.01587 |
| 10 | features.module.24.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01980 | -0.00022 |    0.01580 |
| 11 | features.module.26.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01972 | -0.00014 |    0.01574 |
| 12 | features.module.28.weight | (512, 512, 3, 3) |       2359296 |        2359296 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.01972 | -0.00031 |    0.01574 |
| 13 | classifier.weight         | (10, 512)        |          5120 |           5120 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.06011 | -0.00005 |    0.04519 |
| 14 | Total sparsity:           | -                |      14715584 |       14715584 |    0.00000 |    0.00000 |  0.00000 |  0.00000 |  0.00000 |    0.00000 | 0.00000 |  0.00000 |    0.00000 |
+----+---------------------------+------------------+---------------+----------------+------------+------------+----------+----------+----------+------------+---------+----------+------------+
Total sparsity: 0.00

--- validate (epoch=6)-----------
5000 samples (256 per mini-batch)
==> Top1: 73.200    Top5: 97.760    Loss: 0.777

==> Best [Top1: 73.200   Top5: 97.760   Sparsity:0.00   NNZ-Params: 14715584 on epoch: 6]
Saving checkpoint to: logs/2019.12.13-181650/checkpoint.pth.tar
```

**Comment by cindybrain** [comment]
@nzmora  Really thanks for your quick reply.  I solve the prolem by re-cloning the project !  I think there may be something wrong with my YAML file in on my computer.
